vpr/src/route/segment_stats.cpp

```diff
@@ -68,10 +80,54 @@ void get_segment_usage_stats(std::vector<t_segment_inf>& segment_inf) {
             else
                 length = LONGLINE;
             const short& inode_capacity = rr_graph.node_capacity(rr_id);
-            seg_occ_by_length[length] += route_ctx.rr_node_route_inf[inode].occ();
+            int occ = route_ctx.rr_node_route_inf[inode].occ();
+            seg_occ_by_length[length] += occ;
             seg_cap_by_length[length] += inode_capacity;
-            seg_occ_by_type[seg_type] += route_ctx.rr_node_route_inf[inode].occ();
+            seg_occ_by_type[seg_type] += occ;
             seg_cap_by_type[seg_type] += inode_capacity;
+            VTR_ASSERT(node_type == CHANX|| node_type == CHANY);
+            auto ax = (node_type == CHANX) ? X_AXIS : Y_AXIS;
+            VTR_ASSERT(occ <= 1 && inode_capacity <= 1);
+            directed_occ_by_length[ax][length] += occ;
+            directed_cap_by_length[ax][length] += inode_capacity;
+        }
+    }
+
+    VTR_LOG("\n");
+    VTR_LOG("Total Number of Wiring Segments by Direction: direction length number\n");
+    VTR_LOG("                                              --------- ------ -------\n");
+    for (length = 1; length <= max_segment_length; length++) {
+        for(auto ax : {X_AXIS, Y_AXIS}) {
+            std::string ax_name = (ax==X_AXIS) ? "X" : "Y";
+            if (directed_cap_by_length[ax][length] != 0) {
+                VTR_LOG("                                              %s %4d %6d\n", ax_name.c_str(),
+                        length,
+                        directed_cap_by_length[ax][length]);
```

Comment:
Probably want to add code to print LONGLINE usage (index is the defined constant LONGLINE, which is index 0) if it exists.  Code is later in the routine; can just copy.  Longline isn't used much anymore, but probably good to keep it working since it isn't hard.